addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -282,27 +290,65 @@ function factory(dependencies) {
             const moderated_channel_ids = this.thread.moderation
                 ? [this.thread.id]
                 : undefined;
-            const messages = await this.async(() =>
-                this.env.models['mail.message'].performRpcMessageFetch(
-                    domain,
-                    limit,
-                    moderated_channel_ids,
-                    context,
-                )
-            );
-            this.update({
-                fetchedMessages: [['link', messages]],
-                isLoaded: true,
-                isLoading: false,
-            });
-            if (!extraDomain && messages.length < limit) {
-                this.update({ isAllHistoryLoaded: true });
+            try {
+                const messages = await this.async(() =>
+                    this.env.models['mail.message'].performRpcMessageFetch(
+                        domain,
+                        limit,
+                        moderated_channel_ids,
+                        context,
+                    )
+                );
+                this.update({
+                    fetchedMessages: [['link', messages]],
+                    hasLoadingFailed: false,
+                    isLoaded: true,
+                    isLoading: false,
+                });
+                if (!extraDomain && messages.length < limit) {
+                    this.update({ isAllHistoryLoaded: true });
+                }
+                this.env.messagingBus.trigger('o-thread-cache-loaded-messages', {
+                    fetchedMessages: messages,
+                    threadCache: this,
+                });
+                return messages;
+            } catch(e) {
+                this.update({
+                    hasLoadingFailed: true,
+                    isLoading: false,
+                });
             }
-            this.env.messagingBus.trigger('o-thread-cache-loaded-messages', {
-                fetchedMessages: messages,
-                threadCache: this,
+        }
+
+        /**
+         * Handles the retry of message fetch when failed.
+         * @private
+         */
+        _onHasLoadingFailedChanged() {
+            if (this.hasLoadingFailed) {
+                setTimeout(() => this.update({hasLoadingFailed:false}), 15000);
+            }
+        }
+
+        /**
+         * Loads this thread cache, by fetching the most recent messages in this
+         * conversation.
+         *
+         * @private
+         */
+        _onHasToLoadMessagesChanged() {
+            if (!this.hasToLoadMessages) {
+                return;
+            }
+            this._loadMessages().then(fetchedMessages => {
+                if (!fetchedMessages) {
+                    return;
+                }
+                for (const threadView of this.threadViews) {
+                    threadView.addComponentHint('messages-loaded', { fetchedMessages });
+                }
```

Comment:
This doesn't make sense outside of `_loadMessages`. Adding component hints is a concern of `_loadMessages`.

Reply:
Then you need to address that to @seb-odoo as it's his code. I just added the !fetchedMessages guard.
I don't know how I did that but I duplicated the method, I just removed it.